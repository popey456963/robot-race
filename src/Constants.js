export const MOVE_ONE = 'MOVE_ONE'
export const MOVE_TWO = 'MOVE_TWO'
export const MOVE_THREE = 'MOVE_THREE'
export const BACK_UP = 'BACK_UP'
export const ROTATE_RIGHT = 'ROTATE_RIGHT'
export const ROTATE_LEFT = 'ROTATE_LEFT'
export const U_TURN = 'U_TURN'

export const NORTH = 'NORTH'
export const EAST = 'EAST'
export const SOUTH = 'SOUTH'
export const WEST = 'WEST'

export const PLAIN = 'PLAIN'
export const FLAG = 'FLAG'
export const HOLE = 'HOLE'
export const EXPRESS_CONVEYOR = 'EXPRESS_CONVEYOR'
export const CONVEYOR = 'CONVEYOR'

export const CONVEYOR_INDEX = 1
export const FLAG_INDEX = 2
export const EXPRESS_CONVEYOR_INDEX = 3
export const PLAIN_INDEX = 0

// export const TILE_SCALE = 16
// export const TILE_OFFSET = 6.4
// export const TILE_SIZE = 186

