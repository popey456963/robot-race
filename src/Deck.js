import { MOVE_ONE, MOVE_TWO, MOVE_THREE, BACK_UP, ROTATE_RIGHT, ROTATE_LEFT, U_TURN } from './Constants'

export function createCardType(type, n) {
  return new Array(n).fill(null).map(() => ({ type, priority: 0 }))
}

export function interleave(a, b) {
  // expects same number in both arrays
  return a.reduce((acc, cur, index) => acc.concat(cur, b[index]), [])
}

export default class Deck {
  constructor (ctx, lockedCardPriorities = []) {

    // uturn : 10-60
    // rotate_left, rotate_right: 70 - 420
    // backup: 430 - 480
    // move1: 490 - 660
    // move2: 670 - 780
    // move3: 790-840
    this.cards = [
      ...createCardType(U_TURN, 6),
      ...interleave(createCardType(ROTATE_RIGHT, 18), createCardType(ROTATE_LEFT, 18)),

      ...createCardType(BACK_UP, 6),
      ...createCardType(MOVE_ONE, 18),
      ...createCardType(MOVE_TWO, 12),
      ...createCardType(MOVE_THREE, 6),
    ]

    for (let i = 0; i < this.cards.length; i++) {
      this.cards[i].priority = (i + 1) * 10
    }
    this.cards = ctx.random.Shuffle(this.cards)
    this.cards.filter(card => !(card.priority in lockedCardPriorities))
  }

  drawCard() {
    if (this.cards.length === 0) {
      throw new Error('Cannot draw a card from the deck.')
    }
    return this.cards.pop()
  }
}
