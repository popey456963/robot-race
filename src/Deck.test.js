import Deck, { createCardType, interleave } from './Deck'
import { MOVE_ONE, U_TURN, BACK_UP } from './Constants'

// not a true shuffle
const ctx = {
    random: {
        Shuffle: a => a
    }
}

test('create card type', () => {
    expect(createCardType(MOVE_ONE, 1)).toMatchSnapshot()
    expect(createCardType(U_TURN, 2)).toMatchSnapshot()
    expect(createCardType(BACK_UP, 0)).toMatchSnapshot()
})

test('interleave', () => {
    expect(interleave(['a', 'b'], ['c', 'd'])).toMatchSnapshot()
})

test('deck', () => {
    const deck = new Deck(ctx)
    expect(deck.cards).toMatchSnapshot()

    const deck2 = new Deck(ctx, [10, 50, 100])
    expect(deck2.cards).toMatchSnapshot()

    expect(deck2.drawCard()).toMatchSnapshot()
    expect(deck2.drawCard()).toMatchSnapshot()
    expect(deck2.drawCard()).toMatchSnapshot()

    expect(deck2.cards).toMatchSnapshot()

    deck2.cards = []

    expect(() => deck2.drawCard()).toThrow()
})