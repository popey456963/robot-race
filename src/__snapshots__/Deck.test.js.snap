// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create card type 1`] = `
Array [
  Object {
    "priority": 0,
    "type": "MOVE_ONE",
  },
]
`;

exports[`create card type 2`] = `
Array [
  Object {
    "priority": 0,
    "type": "U_TURN",
  },
  Object {
    "priority": 0,
    "type": "U_TURN",
  },
]
`;

exports[`create card type 3`] = `Array []`;

exports[`deck 1`] = `
Array [
  Object {
    "priority": 10,
    "type": "U_TURN",
  },
  Object {
    "priority": 20,
    "type": "U_TURN",
  },
  Object {
    "priority": 30,
    "type": "U_TURN",
  },
  Object {
    "priority": 40,
    "type": "U_TURN",
  },
  Object {
    "priority": 50,
    "type": "U_TURN",
  },
  Object {
    "priority": 60,
    "type": "U_TURN",
  },
  Object {
    "priority": 70,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 80,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 90,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 100,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 110,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 120,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 130,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 140,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 150,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 160,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 170,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 180,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 190,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 200,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 210,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 220,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 230,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 240,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 250,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 260,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 270,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 280,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 290,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 300,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 310,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 320,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 330,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 340,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 350,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 360,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 370,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 380,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 390,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 400,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 410,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 420,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 430,
    "type": "BACK_UP",
  },
  Object {
    "priority": 440,
    "type": "BACK_UP",
  },
  Object {
    "priority": 450,
    "type": "BACK_UP",
  },
  Object {
    "priority": 460,
    "type": "BACK_UP",
  },
  Object {
    "priority": 470,
    "type": "BACK_UP",
  },
  Object {
    "priority": 480,
    "type": "BACK_UP",
  },
  Object {
    "priority": 490,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 500,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 510,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 520,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 530,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 540,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 550,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 560,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 570,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 580,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 590,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 600,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 610,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 620,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 630,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 640,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 650,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 660,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 670,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 680,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 690,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 700,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 710,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 720,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 730,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 740,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 750,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 760,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 770,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 780,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 790,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 800,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 810,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 820,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 830,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 840,
    "type": "MOVE_THREE",
  },
]
`;

exports[`deck 2`] = `
Array [
  Object {
    "priority": 10,
    "type": "U_TURN",
  },
  Object {
    "priority": 20,
    "type": "U_TURN",
  },
  Object {
    "priority": 30,
    "type": "U_TURN",
  },
  Object {
    "priority": 40,
    "type": "U_TURN",
  },
  Object {
    "priority": 50,
    "type": "U_TURN",
  },
  Object {
    "priority": 60,
    "type": "U_TURN",
  },
  Object {
    "priority": 70,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 80,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 90,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 100,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 110,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 120,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 130,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 140,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 150,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 160,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 170,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 180,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 190,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 200,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 210,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 220,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 230,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 240,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 250,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 260,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 270,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 280,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 290,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 300,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 310,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 320,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 330,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 340,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 350,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 360,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 370,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 380,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 390,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 400,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 410,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 420,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 430,
    "type": "BACK_UP",
  },
  Object {
    "priority": 440,
    "type": "BACK_UP",
  },
  Object {
    "priority": 450,
    "type": "BACK_UP",
  },
  Object {
    "priority": 460,
    "type": "BACK_UP",
  },
  Object {
    "priority": 470,
    "type": "BACK_UP",
  },
  Object {
    "priority": 480,
    "type": "BACK_UP",
  },
  Object {
    "priority": 490,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 500,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 510,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 520,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 530,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 540,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 550,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 560,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 570,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 580,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 590,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 600,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 610,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 620,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 630,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 640,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 650,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 660,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 670,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 680,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 690,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 700,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 710,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 720,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 730,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 740,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 750,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 760,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 770,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 780,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 790,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 800,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 810,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 820,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 830,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 840,
    "type": "MOVE_THREE",
  },
]
`;

exports[`deck 3`] = `
Object {
  "priority": 840,
  "type": "MOVE_THREE",
}
`;

exports[`deck 4`] = `
Object {
  "priority": 830,
  "type": "MOVE_THREE",
}
`;

exports[`deck 5`] = `
Object {
  "priority": 820,
  "type": "MOVE_THREE",
}
`;

exports[`deck 6`] = `
Array [
  Object {
    "priority": 10,
    "type": "U_TURN",
  },
  Object {
    "priority": 20,
    "type": "U_TURN",
  },
  Object {
    "priority": 30,
    "type": "U_TURN",
  },
  Object {
    "priority": 40,
    "type": "U_TURN",
  },
  Object {
    "priority": 50,
    "type": "U_TURN",
  },
  Object {
    "priority": 60,
    "type": "U_TURN",
  },
  Object {
    "priority": 70,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 80,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 90,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 100,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 110,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 120,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 130,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 140,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 150,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 160,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 170,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 180,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 190,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 200,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 210,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 220,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 230,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 240,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 250,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 260,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 270,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 280,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 290,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 300,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 310,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 320,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 330,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 340,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 350,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 360,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 370,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 380,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 390,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 400,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 410,
    "type": "ROTATE_RIGHT",
  },
  Object {
    "priority": 420,
    "type": "ROTATE_LEFT",
  },
  Object {
    "priority": 430,
    "type": "BACK_UP",
  },
  Object {
    "priority": 440,
    "type": "BACK_UP",
  },
  Object {
    "priority": 450,
    "type": "BACK_UP",
  },
  Object {
    "priority": 460,
    "type": "BACK_UP",
  },
  Object {
    "priority": 470,
    "type": "BACK_UP",
  },
  Object {
    "priority": 480,
    "type": "BACK_UP",
  },
  Object {
    "priority": 490,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 500,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 510,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 520,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 530,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 540,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 550,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 560,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 570,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 580,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 590,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 600,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 610,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 620,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 630,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 640,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 650,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 660,
    "type": "MOVE_ONE",
  },
  Object {
    "priority": 670,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 680,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 690,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 700,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 710,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 720,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 730,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 740,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 750,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 760,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 770,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 780,
    "type": "MOVE_TWO",
  },
  Object {
    "priority": 790,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 800,
    "type": "MOVE_THREE",
  },
  Object {
    "priority": 810,
    "type": "MOVE_THREE",
  },
]
`;

exports[`interleave 1`] = `
Array [
  "a",
  "c",
  "b",
  "d",
]
`;
